packages/react-components/src/components/Picker/Picker.tsx

```diff
@@ -0,0 +1,134 @@
+import * as React from 'react';
+import { Error } from '@livechat/design-system-icons/dist/material';
+import { Trigger, TriggerSize } from './Trigger';
+import { IPickerListItem, PickerList } from './PickerList';
+import Icon, { IconSizeName, IconTypeName } from '../Icon';
+
+const baseClass = 'lc-picker';
+export interface IPickerProps {
+  disabled?: boolean;
+  label?: string;
+  error?: string;
+  options: IPickerListItem[];
+  size?: TriggerSize;
+  placeholder?: string;
+  onSelect: (selectedItem: IPickerListItem | null) => void;
+}
+
+export const Picker: React.FC<IPickerProps> = ({
+  disabled,
+  error,
+  label,
+  options,
+  size = TriggerSize.Medium,
+  placeholder = 'Select option',
+  onSelect,
+}) => {
+  const [isListOpen, setIsListOpen] = React.useState<boolean>(false);
+  const [selectedItem, setSelectedItem] =
+    React.useState<IPickerListItem | null>(null);
+  const [searchPhrase, setSearchPhrase] = React.useState<string | null>(null);
+  const triggerRef = React.useRef<HTMLDivElement>(null);
+
+  React.useEffect(() => {
+    if (!isListOpen) {
+      setSearchPhrase(null);
+      return document.removeEventListener('click', onDocumentClick);
+    }
+
+    return document.addEventListener('click', onDocumentClick);
+  }, [isListOpen]);
```

Comment:
I think this is a potential memory leak. Plus returning value from useEffect should be a "clean-up" function.

My proposal:

```
React.useEffect(() => {
   const onDocumentClick = (e) => e.target.contains(triggerRef.current) && isListOpen && setIsListOpen(false)
   
   document.addEventListener('click', onDocumentClick);
   
   return () => document.removeEventListener('click', onDocumentClick)
} [isListOpen])
```

Moreover, @MichalPaszowski had an idea to implement a custom hook `onDocumentClick`, because this pattern is used in a lot of places.

Please double check this potential memory leak.